### PR TITLE
Add `Capsule2d` primitive

### PR DESCRIPTION
**Pull request by Jondolf**
# Objective

Currently, the `Capsule` primitive is technically dimension-agnostic in that it implements both `Primitive2d` and `Primitive3d`. This seems good on paper, but it can often be useful to have separate 2D and 3D versions of primitives.

For example, one might want a two-dimensional capsule mesh. We can't really implement both 2D and 3D meshing for the same type using the upcoming `Meshable` trait (see #11431). We also currently don't implement `Bounded2d` for `Capsule`, see https://github.com/bevyengine/bevy/pull/11336#issuecomment-1890797788.

Having 2D and 3D separate at a type level is more explicit, and also more consistent with the existing primitives, as there are no other types that implement both `Primitive2d` and `Primitive3d` at the same time.

## Solution

Rename `Capsule` to `Capsule3d` and add `Capsule2d`. `Capsule2d` implements `Bounded2d`.

For now, I went for `Capsule2d` for the sake of consistency and clarity. Mathematically the more accurate term would be `Stadium` or `Pill` (see [Wikipedia](https://en.wikipedia.org/wiki/Stadium_(geometry))), but those might be less obvious to game devs. For reference, Godot has [`CapsuleShape2D`](https://docs.godotengine.org/en/stable/classes/class_capsuleshape2d.html). I can rename it if others think the geometrically correct name is better though.

---

## Changelog

- Renamed `Capsule` to `Capsule3d`
- Added `Capsule2d` with `Bounded2d` implemented

### CODE REVIEW

File: crates/bevy_math/src/primitives/dim2.rs [hunk 1]
```diff
@@ -382,6 +382,7 @@ pub struct Rectangle {
     /// Half of the width and height of the rectangle
     pub half_size: Vec2,
 }
+impl Primitive2d for Rectangle {}
```
Comment (Jondolf) [inline crates/bevy_math/src/primitives/dim2.rs:385]: I randomly spotted this missing, lol

**Comment by alice-i-cecile** [comment]
Much prefer `Capsule2D`: the more geometrically correct terms are much less intuitive. If users complain we can add doc aliases.

**Review by alice-i-cecile — APPROVED** [review]

**Review by NiseVoid — APPROVED** [review]
Looks good to me :eyes:

**Review by doonv — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@jondolf a couple of issues following the merge; let me know when you have it fixed up.

**Comment by Jondolf** [comment]
Should be fixed

**Pull request merged**